syntax = "proto3";

package mr;

message Color {
    bool is_rgb = 1;
    bool is_rgba = 2;
    bool is_string = 3;
    uint64 r = 4; 
    uint64 g = 5;
    uint64 b = 6;
    double a = 7;
    string name = 8;
}

message RenderSettings {
    double width = 1;
    double height = 2;
    double padding = 3;
    double stop_radius = 4;
    double line_width = 5;
    
    uint64 bus_label_font_size = 6;
    double bus_label_dx = 7;
    double bus_label_dy = 8;

    uint64 stop_label_font_size = 9;
    double stop_label_dx = 10;
    double stop_label_dy = 11;

    Color underlayer_color = 12;
    double underlayer_width = 13;
    repeated Color palette = 14;
}